core/src/main/java/hera/core/music/TrackScheduler.java

```diff
@@ -0,0 +1,114 @@
+package hera.core.music;
+
+import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
+import com.sedmelluq.discord.lavaplayer.player.event.AudioEventAdapter;
+import com.sedmelluq.discord.lavaplayer.tools.FriendlyException;
+import com.sedmelluq.discord.lavaplayer.track.AudioTrack;
+import com.sedmelluq.discord.lavaplayer.track.AudioTrackEndReason;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TrackScheduler extends AudioEventAdapter {
+
+	private List<AudioTrack> queue = new ArrayList<>();
+
+	private int queueIndex = 0;
+
+	private boolean loopQueue = false;
+
+	private long currentQueueMessageId = 0;
```

Comment:
Not sure why I did this, I'll change it to work the proper way (using a constructor)